File: RQ3/Java-REval-Desc/task27_desc.java
public class FruitDistribution {
    public static void fruit_distribution(String s, int n) {
        java.util.List<Integer> lis = new java.util.ArrayList<>();
        for (String i : s.split(" ")) {
            if (i.matches("\\d+")) {
                lis.add(Integer.parseInt(i));
            }
        }
        int _result = n - lis.stream().mapToInt(Integer::intValue).sum();

        assert func(_result);  // POST: func is equivalent to: In this task, you will be given a string that represents a number of apples and oranges that are distributed in a basket of fruit this basket contains apples, oranges, and mango fruits. Given the string that represents the total number of the oranges and apples and an integer that represent the total number of the fruits in the basket return the number of the mango fruits in the basket. for example: fruit_distribution("5 apples and 6 oranges", 19) ->19 - 5 - 6 = 8 fruit_distribution("0 apples and 1 oranges",3) -> 3 - 0 - 1 = 2 fruit_distribution("2 apples and 3 oranges", 100) -> 100 - 2 - 3 = 95 fruit_distribution("100 apples and 1 oranges",120) -> 120 - 100 - 1 = 19
    }
}

File: RQ3/C-REval-Desc/task27_desc.c
void fruit_distribution(char *s, int n) {
    int lis[100]; // Assuming maximum 100 integers can be in the string
    int count = 0;
    char *token = strtok(s, " ");

    while (token != NULL) {
        int num;
        if (sscanf(token, "%d", &num) == 1) {
            lis[count++] = num;
        }
        token = strtok(NULL, " ");
    }

    int sum = 0;
    for (int i = 0; i < count; i++) {
        sum += lis[i];
    }

    int _result = n - sum;

    assert(func(_result)); // POST: Equivalent condition
}